server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueConfigurationChangeTest.java

```diff
@@ -0,0 +1,308 @@
+
+        CassandraBlobsDAO blobsDAO = new CassandraBlobsDAO(cassandra.getConf(), CassandraConfiguration.DEFAULT_CONFIGURATION, BLOB_ID_FACTORY);
+        mimeMessageStore = MimeMessageStore.factory(blobsDAO).mimeMessageStore();
+        URI amqpUri = new URIBuilder()
+            .setScheme("amqp")
+            .setHost(rabbitMQ.getHostIp())
+            .setPort(rabbitMQ.getPort())
+            .build();
+        URI rabbitManagementUri = new URIBuilder()
+            .setScheme("http")
+            .setHost(rabbitMQ.getHostIp())
+            .setPort(rabbitMQ.getAdminPort())
+            .build();
+        RabbitMQConfiguration rabbitMQConfiguration = RabbitMQConfiguration.builder()
+            .amqpUri(amqpUri)
+            .managementUri(rabbitManagementUri)
+            .build();
+        clock = mock(Clock.class);
+        when(clock.instant()).thenReturn(IN_SLICE_1);
+        random = ThreadLocalRandom.current();
+
+        RabbitMQConnectionFactory rabbitMQConnectionFactory = new RabbitMQConnectionFactory(rabbitMQConfiguration,
+            new AsyncRetryExecutor(Executors.newSingleThreadScheduledExecutor()));
+
+        rabbitClient = new RabbitClient(new RabbitChannelPool(rabbitMQConnectionFactory));
+        mqManagementApi = new RabbitMQManagementApi(rabbitManagementUri, new RabbitMQManagementCredentials("guest", "guest".toCharArray()));
+    }
+
+    private RabbitMQMailQueue getRabbitMQMailQueue(CassandraCluster cassandra, CassandraMailQueueViewConfiguration mailQueueViewConfiguration) throws Exception {
+        MailQueueView mailQueueView = CassandraMailQueueViewTestFactory.factory(clock, random, cassandra.getConf(), cassandra.getTypesProvider(),
+            mailQueueViewConfiguration)
+            .create(MailQueueName.fromString(SPOOL));
+
+        RabbitMQMailQueue.Factory factory = new RabbitMQMailQueue.Factory(
+            new NoopMetricFactory(),
+            rabbitClient,
+            mimeMessageStore,
+            BLOB_ID_FACTORY,
+            mailQueueView,
+            clock);
+        RabbitMQMailQueueFactory mailQueueFactory = new RabbitMQMailQueueFactory(rabbitClient, mqManagementApi, factory);
+        return mailQueueFactory.createQueue(SPOOL);
+    }
+
+    @Test
+    void increasingBucketCountShouldAllowBrowsingAllQueueElements(CassandraCluster cassandra) throws Exception {
+        RabbitMQMailQueue mailQueue = getRabbitMQMailQueue(cassandra, DEFAULT_CONFIGURATION);
+
+        enqueueMailsInSlice(mailQueue, 1, 10);
+
+
+        RabbitMQMailQueue mailQueueWithMoreBuckets = getRabbitMQMailQueue(cassandra,
+            CassandraMailQueueViewConfiguration.builder()
+                    .bucketCount(THREE_BUCKET_COUNT + 2)
+                    .updateBrowseStartPace(UPDATE_BROWSE_START_PACE)
+                    .sliceWindow(ONE_HOUR_SLICE_WINDOW)
+                    .build());
+
+        enqueueMailsInSlice(mailQueueWithMoreBuckets, 2, 10);
```

Comment:
I already commented that code in another PR, please take those comments into account here too.

Reply:
I think I applied the changes requested here

Reply:
see https://github.com/linagora/james-project/pull/1755#discussion_r220836146

I'd like https://github.com/linagora/james-project/pull/1755 to be merged before this one so that we can use MutableClock and others enhancements in this PR.